admin/includes/modules/copy_product_confirm.php

```diff
@@ -47,52 +41,56 @@
-                          '" . (int)$product->fields['products_tax_class_id'] . "',
-                          '" . (int)$product->fields['manufacturers_id'] . "',
-                          '" . zen_db_input(($product->fields['products_quantity_order_min'] == 0 ? 1 : $product->fields['products_quantity_order_min'])) . "',
-                          '" . zen_db_input(($product->fields['products_quantity_order_units'] == 0 ? 1 : $product->fields['products_quantity_order_units'])) . "',
-                          '" . zen_db_input($product->fields['products_priced_by_attribute']) . "',
-                          '" . (int)$product->fields['product_is_free'] . "',
-                          '" . (int)$product->fields['product_is_call'] . "',
-                          '" . (int)$product->fields['products_quantity_mixed'] . "',
-                          '" . zen_db_input($product->fields['product_is_always_free_shipping']) . "',
-                          '" . zen_db_input($product->fields['products_qty_box_status']) . "',
-                          '" . zen_db_input($product->fields['products_quantity_order_max']) . "',
-                          '" . zen_db_input($product->fields['products_sort_order']) . "',
-                          '" . zen_db_input($product->fields['products_price_sorter']) . "',
-                          '" . zen_db_input($categories_id) . "')");
+        $sql_data_array = array(); 
+        $separately_updated_fields = array( 
+          'products_id',
+          'products_last_modified',
+          'products_date_added',
+          'products_date_available',
+        ); 
+        $casted_fields = array( 
+          'products_quantity' =>  'float', 
+          'products_price' =>  'float', 
+          'products_weight' =>  'float',
+          'products_tax_class_id' =>  'int', 
+          'manufacturers_id' =>  'int', 
+          'product_is_free' =>  'int', 
+          'product_is_call' =>  'int', 
+          'products_quantity_mixed' =>  'int', 
+        ); 
+        foreach ($product->fields as $key => $value) {
+          $value = zen_db_input($value); 
+          if (in_array($key, $separately_updated_fields)) {
+            continue; 
+          }
+          if (array_key_exists($key, $casted_fields)) {
+            if ($casted_fields[$key] == 'int') { 
+              $sql_data_array[$key] = (int)$value; 
+            } else {
+              $sql_data_array[$key] = (!zen_not_null($value) || $value == '' || $value == 0) ? 0 : $value; 
+            }
+          } else { 
+            $sql_data_array[$key] = $value; 
+          }
+        }
+
+        // separately_updated_fields - last_modified and products_id are skipped
+        $sql_data_array['products_date_added'] = 'now()';
+        $sql_data_array['products_date_available'] = (!empty(zen_db_input($product->fields['products_date_available'])) ? zen_db_input($product->fields['products_date_available']) : 'null'); 
```

Comment:
This could be simplified slightly by removing the first call to zen_db_input during the conditional check:
```diff
-  $sql_data_array['products_date_available'] = (!empty(zen_db_input($product->fields['products_date_available'])) ? zen_db_input($product->fields['products_date_available']) : 'null'); 
+  $sql_data_array['products_date_available'] = (!empty($product->fields['products_date_available']) ? zen_db_input($product->fields['products_date_available']) : 'null'); 
```